server/conf.c

```diff
@@ -116,44 +116,40 @@ static void ups_update(const char *fn, const char *name, const char *desc)
 /* return 1 if usable, 0 if not */
 static int parse_upsd_conf_args(int numargs, char **arg)
 {
+	unsigned int	temp;
+
 	/* everything below here uses up through arg[1] */
 	if (numargs < 2)
 		return 0;
 
 	/* MAXAGE <seconds> */
 	if (!strcmp(arg[0], "MAXAGE")) {
-		if (isdigit(arg[1])) {
-			maxage = atoi(arg[1]);
+		if (str_to_uint(arg[1], &temp, 10)) {
+			maxage = temp;
 			return 1;
 		}
-		else {
-			upslogx(LOG_ERR, "MAXAGE has non numeric value (%s)!", arg[1]);
-			return 0;
-		}
+		upslogx(LOG_ERR, "Could not convert MAXAGE (%s) to an unsigned int (%s)!", arg[1], strerror(errno));
```

Comment:
now that I see it again... perhaps, to avoid retyping the options, we could use something like this:
```
upslogx(LOG_ERR, "Could not convert %s (%s) to an unsigned int (%s)!", arg[0], arg[1], strerror(errno));
```